### PR TITLE
Kaldi egs

### PR DESCRIPTION
**Pull request by jcftang**
the egs directory contains a number of scripts and recipes that kaldi users tend to use, this PR just install all of it into the prefix of kaldi

### CODE REVIEW

File: var/spack/repos/builtin/packages/kaldi/package.py [hunk 1]
```diff
@@ -69,6 +72,9 @@ def install(self, spec, prefix):
             configure_args.append('--use-cuda=yes')
             configure_args.append('--cudatk-dir=' + spec['cuda'].prefix)
 
+        if spec.satisfies('@2019-07-29:'):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/kaldi/package.py]: @jcftang if this version range is open-ended then the `depends_on... when=...` for `cub` should be open-ended.

Comment (jcftang) [reply var/spack/repos/builtin/packages/kaldi/package.py]: you mean doing

```
depends_on('cub', when='@2019-07-29:')
```

and have newer versions of kaldi depend on cub...

Comment (scheibelp) [reply var/spack/repos/builtin/packages/kaldi/package.py]: Yes: otherwise, versions after `@2019-07-29` (when they are added) would run into an error here

Comment (jcftang) [reply var/spack/repos/builtin/packages/kaldi/package.py]: this is done

File: var/spack/repos/builtin/packages/kaldi/package.py [hunk 2]
```diff
@@ -35,6 +36,8 @@ class Kaldi(Package):    # Does not use Autotools
     depends_on('speex', type='run')
     depends_on('openfst@1.4.1-patch', when='@2015-10-07')
     depends_on('openfst@1.6.0:', when='@2018-07-11')
+    depends_on('openfst@1.6.0:', when='@2019-07-29')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/kaldi/package.py]: @jcftang Apologies but I think there will be a similar issue here: versions of `kaldi` after `2019-07-29` will also want to use `openfst@1.6.0:` I assume? I think this should update the `when` to be `@2019-07-29:`. If in fact you anticipate that later versions will not need that constraint, let me know (in which case LGTM).

Comment (jcftang) [reply var/spack/repos/builtin/packages/kaldi/package.py]: Yea, I wasn't too sure of this one either, but openfst updates have been pretty rare in the kaldi project afaik, I think this is something that should be addressed as new versions of kaldi are added and released. This is a bit different from the cub package which has a higher update cycle.

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**